Adding stripeClient from @better-auth/stripe/client to auth client removes twoFactorClient types from $ERROR_CODES

### Is this suited for github?

- [x] Yes, this is suited for github

### To Reproduce

When using the `@better-auth/stripe` plugin alongside the `twoFactorClient` plugin, the $ERROR_CODES type no longer includes the two-factor authentication error codes. This appears to be a type composition/merging issue between plugins.

1. Create an auth client with only the `twoFactorClient` plugin:

```typescript
import { createAuthClient } from "better-auth/react";
import { twoFactorClient } from "better-auth/client/plugins";

const authClient = createAuthClient({
    baseURL: "http://localhost:3000",
    basePath: "/api/auth",
    plugins: [
        twoFactorClient(),
    ],
});

export type AuthErrorCode = keyof (typeof authClient)["$ERROR_CODES"];
``` 

2. Use the `AuthErrorCode` type to have type-safe translations for all errors:

```typescript
const AUTH: Record<AuthErrorCode, string> = {
    // ... other error codes
    OTP_HAS_EXPIRED: "The code has expired",
    INVALID_TWO_FACTOR_COOKIE: "Invalid two-factor cookie",
    TWO_FACTOR_NOT_ENABLED: "Two-factor is not enabled",
    // ... etc - all two-factor error codes work fine
};
``` 

3. Add the `stripeClient` plugin

```typescript
import { createAuthClient } from "better-auth/react";
import { twoFactorClient } from "better-auth/client/plugins";
import { stripeClient } from "@better-auth/stripe/client";

const authClient = createAuthClient({
    baseURL: "http://localhost:3000",
    basePath: "/api/auth",
    plugins: [
        twoFactorClient(),
        stripeClient({
            subscription: true,
        }),
    ],
});

export type AuthErrorCode = keyof (typeof authClient)["$ERROR_CODES"];
``` 

4. TypeScript now errors on two-factor error codes:

`error TS2353: Object literal may only specify known properties, and 'OTP_HAS_EXPIRED' does not exist in type 'Record<...>'`

### Current vs. Expected behavior

**Current Behavior**

When `stripeClient` is added to the plugins array, the `$ERROR_CODES` type no longer includes error codes from `twoFactorClient` such as:

- `OTP_HAS_EXPIRED`
- `OTP_NOT_ENABLED`
- `INVALID_BACKUP_CODE`
- `INVALID_TWO_FACTOR_COOKIE`
- `TOTP_NOT_ENABLED`
- `TOO_MANY_ATTEMPTS_REQUEST_NEW_CODE`
- `TWO_FACTOR_NOT_ENABLED`
- `BACKUP_CODES_NOT_ENABLED`
- `INVALID_CODE`

**Expected Behavior**

The `$ERROR_CODES` type should be a union of all type and adding `stripeClient` plugin should not remove/invalidate twoFactorClient error codes from the type.

### What version of Better Auth are you using?

1.3.8

### System info

```bash
{
  "system": {
    "platform": "darwin",
    "arch": "arm64",
    "version": "Darwin Kernel Version 25.0.0: Wed Sep 17 21:41:45 PDT 2025; root:xnu-12377.1.9~141/RELEASE_ARM64_T6000",
    "release": "25.0.0",
    "cpuCount": 10,
    "cpuModel": "Apple M1 Pro",
    "totalMemory": "16.00 GB",
    "freeMemory": "0.12 GB"
  },
  "node": {
    "version": "v22.14.0",
    "env": "development"
  },
  "packageManager": {
    "name": "npm",
    "version": "11.6.1"
  },
  "frameworks": [
    {
      "name": "next",
      "version": "^15.5.0"
    },
    {
      "name": "react",
      "version": "19.1"
    }
  ],
  "databases": null,
  "betterAuth": {
    "version": "^1.3.8",
    "config": null
  }
}
```

### Which area(s) are affected? (Select all that apply)

Types

### Auth config (if applicable)

```typescript

```

### Additional context

_No response_

--- a/packages/stripe/src/index.ts
+++ b/packages/stripe/src/index.ts
@@ -1399,6 +1399,7 @@ export const stripe = <O extends StripeOptions>(options: O) => {
 			};
 		},
 		schema: getSchema(options),
+		$ERROR_CODES: STRIPE_ERROR_CODES,
 	} satisfies BetterAuthPlugin;
 };
 

